azkaban-exec-server/src/main/java/azkaban/execapp/reporter/AzkabanEventReporter.java

```diff
@@ -0,0 +1,20 @@
+package azkaban.execapp.reporter;
+
+import azkaban.event.Event;
+import java.util.Map;
+
+/**
+ * Implement this interface to report flow and job events. Event reporter
+ * can be turned on by setting the property 'event.reporter.enabled' to true.
+ *
+ * By default, a KafkaAvroEventReporter is provided. Alternate implementations
+ * can be provided by setting the property 'event.reporter.class'.
+ *
+ * The constructor will be called with a azkaban.utils.Props object passed as
+ * the only parameter. If such a constructor doesn't exist, then the AzkabanEventReporter
+ * instantiation will fail.
+ */
+public interface AzkabanEventReporter {
+
+  boolean report(Event event, Map<String, String> metadata);
```

Comment:
Should the content of metadata be defined? Or is it the intention that the metadata should be treated as a black box by the reporter?

Reply:
Ok, let me check on these. I will make the suggested changes.

Reply:
Metadata can be treated as a black-box though.

Reply:
Ok, I have created a separate event interface for this purpose and moved the event interface and AzkabanEventReporter interface to the azkaban.spi module.